spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -296,71 +310,83 @@ public void run() {
 				// Always receive messages in batch mode.
 				try {
 					if (!inRetryMode.get() && !messagesPendingInBatch.get()) {
+						DefaultPulsarMessageListenerContainer.this.receiveInProgress = true;
 						messages = this.consumer.batchReceive();
+						DefaultPulsarMessageListenerContainer.this.receiveInProgress = false;
 					}
 				}
 				catch (PulsarClientException e) {
-					DefaultPulsarMessageListenerContainer.this.logger.error(e, () -> "Error receiving messages.");
-				}
-				Assert.isTrue(messages != null, "Messages cannot be null.");
-				if (this.isBatchListener) {
-					if (!inRetryMode.get() && !messagesPendingInBatch.get()) {
-						messageList = new ArrayList<>();
-						messages.forEach(messageList::add);
+					if (e.getCause() instanceof InterruptedException) {
+						DefaultPulsarMessageListenerContainer.this.logger.debug(e, () -> "Error receiving messages.");
```

Comment:
I'm torn as to what level of logging we should use in this case. Would we want to ever see this message? How often would this be happening and be an expected thing? 

Regardless of level, I think we should differentiate this from the other error case w/ a more descriptive message about what is going on.

Reply:
Ok, I will clarify the log message. I was also struggling to decide what level should we log. We don't want this message to appear in the logs causing confusion to the user, but should be available for debugging purposes, thus went with `debug`.